xds/internal/xdsclient/controller/controller.go

```diff
@@ -88,6 +88,17 @@ type Controller struct {
 	lrsClients map[string]*lrsClient
 }
 
+var grpcDial = grpc.Dial
+
+// SetDialInterceptor sets an interceptor for the controller. The interceptor
+// can be used to manipulate the dial options or change the target if needed.
+// The SetDialInterceptor must be called before gRPC to make sure it affects
+// all the controllers created.
+// To reset any dial interceptor set, pass in grpc.Dial as the parameter.
+func SetDialInterceptor(interceptor func(target string, opts ...grpc.DialOption) (*grpc.ClientConn, error)) {
```

Comment:
This is not really setting an "interceptor".
Just `SetGRPCDial`?

Reply:
Done.